cmd/snap/cmd_quota.go

```diff
@@ -165,6 +174,26 @@ func parseCpuQuota(cpuMax string) (count int, percentage int, err error) {
 	return count, percentage, nil
 }
 
+func parseJournalRateQuota(journalRateLimit string) (count int, period time.Duration, err error) {
+	// the rate limit is a string of the form N/P, where N is the number of
+	// messages and P is the period as a time string (e.g 5s)
+	parts := strings.Split(journalRateLimit, "/")
+	if len(parts) != 2 {
+		return 0, 0, fmt.Errorf("rate limit must be of the form <number of messages>/<period duration>")
+	}
+
+	count, err = strconv.Atoi(parts[0])
+	if err != nil {
+		return 0, 0, err
```

Comment:
(nitpick) I would not return a "naked" error here, it will probably be hard to read, maybe similar to below, something like: `cannot parse number of message: %v` ? I noticed in the unit test for this that it reads nicely so feel free to ignore this nitpick.

Reply:
Good point, i'll change this!